File: hw/caizapanta/u3/HW19Composite/Composite/src/main/java/ec/edu/espe/composite/model/Supervisor.java
package ec.edu.espe.composite.model;

import java.util.ArrayList;

/**
 *
 * @author Tammy Caizapanta SoftCrafters DCCO ESPE
 */
public abstract class Supervisor extends Employee {

    // Changed from Vector to ArrayList
    protected ArrayList<Employee> directReports = new ArrayList<>();

    public void stateName() {
        super.stateName(); // print name of this employee first
        if (directReports.size() > 0) // be sure there are elements
        {
            // Changed to enhanced for-loop for ArrayList
            for (Employee emp : directReports) {
                emp.stateName();
            }
        }
    }

    public void add(Employee anEmployee) {
        this.directReports.add(anEmployee); // Changed from addElement to add
    }
}

File: hw/fuentes/U3/HW20COMPOSITE/HW20COMPOSITEPY/ec/edu/espe/composite/model/Supervisor.py
from model.Employee import Employee

class Supervisor(Employee):
    def __init__(self):
        super().__init__()
        self.directReports = []

    def add(self, anEmployee):
        self.directReports.append(anEmployee)

    def stateName(self):
        super().stateName()
        for emp in self.directReports:
            emp.stateName()
